Source/Core/DolphinQt/SearchBar.cpp

```diff
@@ -34,20 +38,35 @@ void SearchBar::CreateWidgets()
   setLayout(layout);
 }
 
-void SearchBar::Toggle()
+void SearchBar::Show()
 {
-  m_search_edit->clear();
+  m_search_edit->setFocus();
+  m_search_edit->selectAll();
 
-  setHidden(isVisible());
+  show();
+}
+
+void SearchBar::Hide()
+{
+  m_search_edit->clear();
```

Comment:
I don't think the `QLineEdit` should be cleared on close. Just make sure the text is selected when it's reopened.

Reply:
Done.